frontend/test/metabase-lib/lib/Question.unit.spec.js

```diff
@@ -459,34 +458,67 @@ describe("Question", () => {
         const question = orders_count_question
           .setDisplay("funnel")
           .lockDisplay()
-          .maybeUnlockDisplay(["table", "scalar"])
-          .setDefaultDisplay();
+          .maybeResetDisplay(["table", "scalar"]);
 
         expect(question.display()).toBe("scalar");
       });
     });
 
-    describe("maybeUnlockDisplay", () => {
-      it("should keep display locked when it was locked with nonsense display", () => {
+    describe("maybeResetDisplay", () => {
+      it("should do nothing when it was locked with nonsense display", () => {
         const sensibleDisplays = ["table", "scalar"];
         const previousSensibleDisplays = sensibleDisplays;
         const question = new Question(orders_count_card, metadata)
           .setDisplay("funnel")
           .lockDisplay()
-          .maybeUnlockDisplay(sensibleDisplays, previousSensibleDisplays);
+          .maybeResetDisplay(sensibleDisplays, previousSensibleDisplays);
 
         expect(question.displayIsLocked()).toBe(true);
+        expect(question.display()).toBe("funnel");
       });
 
-      it("should unlock display it was locked with sensible display which has become unsensible", () => {
+      it("should use default display when nonsense display is used and was not locked", () => {
+        const sensibleDisplays = ["table", "scalar"];
+        const question = base_question
+          .setDisplay("funnel")
+          .maybeResetDisplay(sensibleDisplays, sensibleDisplays);
+
+        expect(question.display()).not.toBe("funnel");
+        expect(question.display()).toBe("table");
+      });
+
+      it("should unlock and use new sensible display when it was locked with sensible display which has become not sensible", () => {
         const previousSensibleDisplays = ["funnel"];
         const sensibleDisplays = ["table", "scalar"];
         const question = orders_count_question
           .setDisplay("funnel")
           .lockDisplay()
-          .maybeUnlockDisplay(sensibleDisplays, previousSensibleDisplays);
+          .maybeResetDisplay(sensibleDisplays, previousSensibleDisplays);
 
         expect(question.displayIsLocked()).toBe(false);
+        expect(question.display()).not.toBe("funnel");
+        expect(sensibleDisplays).toContain(question.display());
```

Comment:
Semantically, this is inversed - I don't expect array to contain the element - I expect the element to be in an array.

Is there anything similar to `expect(item).toBeIn(array)`?

Reply:
Is there any reason why we don't just test if the display is a certain type i.e. why don't we
```ts
expect(question.display()).not.toBe("table");
```

Reply:
The same explanation applies here: https://github.com/metabase/metabase/pull/31646/files#r1250946190